files/secure_media_views.py

```
@@ -100,6 +104,61 @@
 """
 
 
+def get_media_path_cache_key(file_path: str) -> str:
+    """
+    Generate cache key for file path → Media ID mapping.
+    Uses MD5 hash to keep key length reasonable.
+    """
+    path_hash = hashlib.md5(file_path.encode('utf-8')).hexdigest()[:16]
```

Ruff (0.14.4)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)